src/ui/main_window.py

```
@@ -331,6 +356,26 @@ def _on_roi_selected(self, roi: ROI) -> None:
+        if "display" not in self.experiment.settings:
+            self.experiment.settings["display"] = {}
+        self.experiment.settings["display"]["exposure"] = exposure
+        self.experiment.settings["display"]["contrast"] = contrast
+        # Persist to file if we have a path
+        if self.current_experiment_path:
+            try:
+                self.manager.save_experiment(
+                    self.experiment, self.current_experiment_path
+                )
+            except Exception:
+                pass
```

Ruff (0.14.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)